MyStack.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;  
+namespace Stack
+{
+    public class MyStack<T> : IMyStack<T>{
+        private List<T> stack;
+        public MyStack() 
+        {
+            this.stack = new List<T>();
+        }
+        
+        public void Push(T x) 
+        {
+            stack.Add(x);
+        }
+        
+        public T Pop() 
+        {
+            if(this.IsEmpty())
+            {
+                throw new InvalidOperationException("The stack is empty!");
+            }
+            T top = this.Peak();
+            stack.RemoveAt(stack.Count - 1);
```

Comment:
what do you think about having a private variable for the idx of the last element (top ) of the stack?